mining_test.go

```diff
@@ -6,10 +6,141 @@
 package main
 
 import (
-	"fmt"
+	"container/heap"
+	"math/rand"
 	"testing"
+
+	"github.com/decred/dcrd/blockchain/stake"
 )
 
-func MineTest(t *testing.T) {
-	fmt.Println("Hello World!")
+// TestTxFeePrioHeap tests the priority heaps including the stake types for both
+// transaction fees per KB and transaction priority. It ensures that the primary
+// sorting is first by stake type, and then by the latter chosen priority type.
+func TestTxFeePrioHeap(t *testing.T) {
+	numElements := 1000
+	ph := newTxPriorityQueue(numElements, txPQByStakeAndFee)
+
+	for i := 0; i < numElements; i++ {
+		randType := stake.TxType(rand.Intn(4))
+		randPrio := rand.Float64() * 100
+		randFeePerKB := rand.Float64() * 10
+		prioItem := &txPrioItem{
+			tx:       nil,
+			txType:   randType,
+			priority: randPrio,
+			feePerKB: randFeePerKB,
+		}
+		heap.Push(ph, prioItem)
+	}
+
+	// Test sorting by stake and fee per KB.
+	last := &txPrioItem{
+		tx:       nil,
+		txType:   stake.TxTypeSSGen,
+		priority: 10000.0,
+		feePerKB: 10000.0,
+	}
+	for i := 0; i < numElements; i++ {
+		prioItem := heap.Pop(ph)
+		txpi, ok := prioItem.(*txPrioItem)
+		if ok {
+			if txpi.feePerKB > last.feePerKB &&
```

Comment:
Don't forget to fix these tests.  It should be `>=` on the first condition and `>` on the second condition.  Otherwise this can result in false positives.